google/resource_compute_instance.go

```diff
@@ -964,6 +981,55 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 		d.SetPartial("tags")
 	}
 
+	if d.HasChange("instance_state") {
```

Comment:
Stated above, but we'll need to watch out for changes to the instance state in this method; https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_compute_instance.go#L1253